internal/telemetry/telemetrytest/telemetrytest.go

```diff
@@ -0,0 +1,55 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+// Package telemetrytest provides a mock implementation of the telemetry client for testing purposes
+package telemetrytest
+
+import (
+	"sync"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+)
+
+// MockClient implements Client and is used for testing purposes.
+type MockClient struct {
+	mu              sync.Mutex
+	Started         bool
+	Configuration   []telemetry.Configuration
+	ProfilerEnabled bool
+	AsmEnabled      bool
+}
+
+// Start starts and adds configuration data to the mock client.
+func (c *MockClient) Start(configuration []telemetry.Configuration) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.Started = true
+	c.Configuration = append(c.Configuration, configuration...)
+}
+
+// ProductChange signals that a certain product is enabled or disabled for the mock client.
+func (c *MockClient) ProductChange(namespace telemetry.Namespace, enabled bool, configuration []telemetry.Configuration) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.AsmEnabled = (namespace == telemetry.NamespaceASM) && enabled
```

Comment:
Is it intentional that turning on a Product would turn off the other ones?
If I read this correctly after calling
```
ProductChange(ASM, true,...)
ProductChange(Profilers, true,...)
```
Then ProfilerEnabled will be true, but AsmEnabled will be false?
Probably better off with a switch on the `namespace` and a `default` case that panics to force any test that calls this with an unexpected namespace to fail

Reply:
That makes sense, thanks for catching! Changed to a select statement